components/appState/uri/resolver.js

```diff
@@ -46,19 +60,18 @@ module.exports = function(stateConf, state, aliases, encode) {
         return indexA - indexB;
     });
 
-    _.each(stateKeys, function(name) {
-        var data = state[name];
-
-        if (_.isArray(data)) {
-            _.each(data, function(state) {
-                var type = state.type;
-                var data = _.omit(state, 'type');
-                processState(data, type);
-            });
+    var queryParamsList = stateConf.get('queryParamsList') || [];
+    var queryKeys = [];
+    var mainKeys = [];
+    _.each(stateKeys, function(stateKey) {
```

Comment:
I have updated lodash version to 3.10.1 and bump Slot version to 0.12.0. Now we can use all lodash magic ;)